test/policies/jwt/jwt.test.js

```diff
@@ -0,0 +1,171 @@
+const idGen = require('uuid-base62');
+const fs = require('fs');
+const request = require('supertest');
+const jwt = require('jsonwebtoken');
+
+const db = require('../../../lib/db');
+
+const services = require('../../../lib/services');
+const credentialService = services.credential;
+const userService = services.user;
+
+const serverHelper = require('../../common/server-helper');
+const config = require('../../../lib/config');
+const testHelper = require('../../common/routing.helper')();
+
+const originalGatewayConfig = config.gatewayConfig;
+
+let gateway;
+let backend;
+let jwtCredential;
+
+describe('JWT policy', () => {
+  [{
+    description: 'Secret string',
+    jwtSecret: 'superSecretString',
+    jwtSignOptions: {},
+    actionConfig: {
+      secret: 'superSecretString'
+    }
+  }, {
+    description: 'Secret file',
+    jwtSecret: fs.readFileSync(require.resolve('../../fixtures/certs/client/client.key')),
+    jwtSignOptions: {
+      algorithm: 'RS256'
+    },
+    actionConfig: {
+      secretFile: require.resolve('../../fixtures/certs/client/client.crt'),
+      jwtExtractor: 'query',
+      jwtExtractorField: 'jwtKey'
+    }
+  }].forEach((jwtSecretTestCase) => {
+    describe(jwtSecretTestCase.description, () => {
+      before('setup', () => {
+        config.gatewayConfig = {
+          http: {
+            port: 0
+          },
+          serviceEndpoints: {
+            backend: {
+              url: 'http://localhost:6057'
+            }
+          },
+          apiEndpoints: {
+            api: {
+              host: '*'
+            }
+          },
+          policies: ['jwt', 'proxy'],
+          pipelines: {
+            pipeline1: {
+              apiEndpoints: ['api'],
+              policies: [{
+                'jwt': {
+                  action: jwtSecretTestCase.actionConfig
+                }
+              },
+              {
+                proxy: [
+                  {
+                    action: { serviceEndpoint: 'backend' }
+                  }
+                ]
+              }
+              ]
+            }
+          }
+        };
+
+        return db.flushdb()
+          .then(() => userService.insert({
+            username: idGen.v4(),
+            firstname: 'Vincenzo',
```

Comment:
In general, Can we use some superhero theme or some not real names for tests?

Reply:
Yes, it makes sense. `Clark Kent` will do the job.

We could also emply [faker](https://github.com/marak/Faker.js/) but I do not think we really need this right now.

Reply:
took a look at faker, it is awesome. Eventually we can use it.

Reply:
Well check also this then: it's mind blowind. [json-schema-faker](https://github.com/json-schema-faker/json-schema-faker)